crates/ruff_python_formatter/src/expression/expr_subscript.rs

```diff
@@ -97,7 +101,21 @@ impl NeedsParentheses for ExprSubscript {
                 OptionalParentheses::Never
             } else {
                 match self.value.needs_parentheses(self.into(), context) {
-                    OptionalParentheses::BestFit => OptionalParentheses::Never,
+                    OptionalParentheses::BestFit => {
+                        if parent.as_stmt_function_def().is_some_and(|function_def| {
+                            function_def
+                                .returns
+                                .as_deref()
+                                .and_then(Expr::as_subscript_expr)
+                                == Some(self)
```

Comment:
Can you use `is_some_and` here?

Reply:
I already use `is_some_and` on line 105. Do you mean for the equality comparison? I prefer avoiding lambdas when possible because it leads to very nested code.